test/hotspot/jtreg/serviceability/jvmti/SuspendWithCurrentThread/libSuspendWithCurrentThread.cpp

```diff
@@ -22,13 +22,15 @@
  */
 
 #include <string.h>
+#include <atomic>
 #include "jvmti.h"
 
 extern "C" {
 
 static jvmtiEnv* jvmti = NULL;
 static jthread* threads = NULL;
 static jsize threads_count = 0;
+static std::atomic<bool> is_exited_from_suspend;
```

Comment:
Is use of `std::atomic` permitted in tests?

Reply:
I've never seen any such restrictions in our development guidelines.
In general, there are cases where using atomic vs raw monitors is desirable to avoid over sync in tests when there is a need to stress multi-threading.

Reply:
I don't know anything about such restrictions also. The atomics are required in this test because the thread actually managed to process the suspend request during RawMonitorEnter if it is used and just suspend right before entering the critical section. So RawMonitorEnter/RawMonitorExitt might not show the problem. Also, as Serguei mentioned atomics synchronization might help us in stress tests.